nextcord/types/protocols/base_user.py

```diff
@@ -0,0 +1,75 @@
+# The MIT License (MIT)
+# Copyright (c) 2021-present vcokltfre & tag-epic
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from __future__ import annotations
+
+from logging import getLogger
+from typing import TYPE_CHECKING, Optional
+
+if TYPE_CHECKING:
+    from ...client.state import State
+    from ..snowflake import Snowflake
+
+
+logger = getLogger(__name__)
+
+
+class UserProtocol:
+    __slots__ = (
+        "id",
+        "username",
+        "discriminator",
+        "_avatar",
+        "bot",
+        "system",
+        "_banner",
+        "_accent_color",
+        "_premium_type",
+        "_public_flags",
+        "_state",
+    )
+
+    if TYPE_CHECKING:
+        id: Snowflake
```

Comment:
These are not classvars?

Reply:
Class annotations, this is how python works if you just want to annotate an attribute, instance or class

Reply:
Yeah those are annotations